x-pack/plugins/security_solution/public/common/components/filter_group/index.tsx

```diff
@@ -0,0 +1,315 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  ControlGroupInput,
+  ControlGroupContainer,
+  controlGroupInputBuilder,
+  ControlGroupOutput,
+} from '@kbn/controls-plugin/public';
+import { LazyControlGroupRenderer } from '@kbn/controls-plugin/public';
+import type { PropsWithChildren } from 'react';
+import React, { createContext, useCallback, useEffect, useState, useRef, useMemo } from 'react';
+import { ViewMode } from '@kbn/embeddable-plugin/public';
+import {
+  EuiButton,
+  EuiButtonIcon,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLoadingChart,
+  EuiPopover,
+} from '@elastic/eui';
+import type { Subscription } from 'rxjs';
+import styled from 'styled-components';
+import { cloneDeep, debounce } from 'lodash';
+import { withSuspense } from '@kbn/shared-ux-utility';
+import { useInitializeUrlParam } from '../../utils/global_query_string';
+import { URL_PARAM_KEY } from '../../hooks/use_url_state';
+import type { FilterContextType, FilterGroupProps, FilterUrlFormat } from './types';
+import { useFilterUpdatesToUrlSync } from './hooks/use_filter_update_to_url_sync';
+
+type ControlGroupBuilder = typeof controlGroupInputBuilder;
+
+const ControlGroupRenderer = withSuspense(LazyControlGroupRenderer);
+
+export const FilterContext = createContext<FilterContextType | undefined>(undefined);
+
+const FilterWrapper = styled.div.attrs((props) => ({
+  className: props.className,
+}))`
+  & .euiFilterButton-hasActiveFilters {
+    font-weight: 400;
+  }
+
+  & .controlGroup {
+    min-height: 40px;
+  }
+`;
+
+export const FilterGroup = (props: PropsWithChildren<FilterGroupProps>) => {
+  const {
+    dataViewId,
+    onFilterChange,
+    timeRange,
+    filters,
+    query,
+    chainingSystem = 'HIERARCHICAL',
+    initialControls,
+  } = props;
+
+  const filterChangedSubscription = useRef<Subscription>();
+  const inputChangedSubscription = useRef<Subscription>();
+
+  const [controlGroup, setControlGroup] = useState<ControlGroupContainer>();
+
+  const [controlGroupInputUpdates, setControlGroupInputUpdates] = useState<
+    ControlGroupInput | undefined
+  >();
+
+  const [initialUrlParam, setInitialUrlParam] =
+    useState<Array<FilterUrlFormat[keyof FilterUrlFormat]>>();
+
+  const urlDataApplied = useRef<boolean>(false);
+
+  const [isContextMenuVisible, setIsContextMenuVisible] = useState(false);
+
+  const toggleContextMenu = useCallback(() => {
+    setIsContextMenuVisible((prev) => !prev);
+  }, []);
+
+  const onUrlParamInit = (param: Array<FilterUrlFormat[keyof FilterUrlFormat]> | null) => {
+    if (param == null) return;
+    try {
+      setInitialUrlParam(param);
+    } catch (err) {
+      // if there is an error ignore url Param
+      // eslint-disable-next-line no-console
+      console.error(err);
+    }
+  };
+
+  useInitializeUrlParam(URL_PARAM_KEY.pageFilter, onUrlParamInit);
```

Comment:
I would add a comment to the `URL_PARAM_KEY` in `x-pack/plugins/security_solution/public/common/hooks/use_url_state.ts` essentially saying the initialization and updating logic are in the component itself. I think it'll help in the future if there are any refactors if someone for whatever reason thinks `pageFilters` is not being used

Reply:
Will do. Thanks.